internal/cmd/schema-registry/command_cluster_delete.go

```diff
@@ -44,21 +44,29 @@ func (c *command) clusterDelete(cmd *cobra.Command, _ []string) error {
 		return err
 	}
 
-	cluster, err := c.Context.FetchSchemaRegistryByEnvironmentId(environmentId)
+	clusters, err := c.V2Client.GetSchemaRegistryClustersByEnvironment(environmentId)
 	if err != nil {
 		return err
 	}
 
-	promptMsg := fmt.Sprintf(`Are you sure you want to delete %s "%s" for %s "%s"?`, resource.SchemaRegistryCluster, cluster.Id, resource.Environment, environmentId)
+	if len(clusters) == 0 {
+		return errors.NewSRNotEnabledError()
+	}
+
+	promptMsg := fmt.Sprintf(`Are you sure you want to delete %s "%s" for %s "%s"?`, resource.SchemaRegistryCluster, clusters[0].GetId(), resource.Environment, environmentId)
 	if ok, err := form.ConfirmDeletion(cmd, promptMsg, ""); err != nil || !ok {
 		return err
 	}
 
-	err = c.Client.SchemaRegistry.DeleteSchemaRegistryCluster(cluster)
+	err = c.V2Client.DeleteSchemaRegistryCluster(clusters[0].GetId(), environmentId)
 	if err != nil {
 		return err
 	}
 
+	ctx := c.Config.Context()
+	ctx.SchemaRegistryClusters[environmentId] = nil
```

Comment:
@brianstrauch - AFAIK, neither the HTTP endpoint URL nor the cluster ID ever changes during the lifetime of a SR cluster.

Just for context, are we talking about deleting and recreating the clusters in the same environment here?

Reply:
> Just for context, are we talking about deleting and recreating the clusters in the same environment here?

Yes, we don't want to cache the Schema Registry cluster URL in the CLI if it's possible for a customer to delete and recreate it from the UI (which would change the URL)

Reply:
Makes sense, because that is possible. A customer could delete their cluster via the UI and recreate a cluster in the same environment in another region or a cluster with different physical cluster in the same region, changing the SR cluster url for that env.